storage/innobase/dict/dict0mem.cc

```diff
@@ -1261,6 +1262,15 @@ bool dict_table_t::deserialise_columns(const byte* metadata, ulint len)
 	DBUG_ASSERT(col == &dropped_cols[n_dropped_cols]);
 
 	UT_LIST_GET_FIRST(indexes)->reconstruct_fields();
+
+	rw_lock_x_lock(&index->lock);
+	committed_count_inited = (12 + num_non_pk_fields * 2 == len);
```

Comment:
is there a constant/define from which this 12 is dervied?

Reply:
The metadata blob is laid out as follows: 
4 bytes => `num_non_pk_fields`
2 bytes * `num_non_pk_fields` => field map element for each non-PK field
8 bytes (not always present) => `committed_count`

The 12 is derived from adding the bytes for `num_non_pk_fields` with those for `committed_count`. Should I explicitly define `NUM_NON_PK_FIELDS_SIZE`, `FIELD_MAP_ELEM_SIZE` and `COMMITTED_COUNT_SIZE` constants to clarify the intent behind this code?

Reply:
It would help clarify the intent. If they can be expressions base on the metadata structure that would be even better. If they can be coupled with `compile_time_assert` to ensure that any future change to the metadata blob will require correcting the coding here too.

Reply:
I've defined the field size constants for the metadata blob, but I'm not sure about your `compile_time_assert` suggestion, since the metadata blob has a runtime variable length.